src/Database/LSMTree.hs

```diff
@@ -993,10 +995,12 @@ insert table k v b =
 {- |
 Variant of 'insert' for batch insertions.
 
-The worst-case disk I\/O complexity of this operation depends on the merge policy of the table:
+The worst-case disk I\/O complexity of this operation depends on the merge policy and the merge schedule of the table:
 
-['LazyLevelling']:
-    \(O(b \: \log_T \frac{n}{B})\).
+['LazyLevelling'\/'Incremental']:
+    \(O(b \: \frac{1}{P} \: \log_T \frac{n}{B})\).
+['LazyLevelling'\/'OneShot']:
+    \(O(b \log_T \frac{b}{B} + \frac{n}{P})\).
```

Comment:
And how did you get to $O(b \log_T \frac{b}{B} + \frac{n}{P})$ for the bulk operations?

Reply:
The added `b*logT b/B` term is the cost of inserting the batch, which is only there because it dominates the other term when `b > n`.